File: parse_test.ts
import * as fs from 'fs';
import * as cheerio from 'cheerio';

const html = fs.readFileSync('bot/woningaanbod_dump.html', 'utf-8');
const $ = cheerio.load(html);

const links = $('a[href*="PublicatieId="]');
console.log(`Found ${links.length} links`);

if (links.length > 0) {
  const firstLink = links.first();
  console.log('HTML of first card:', firstLink.parent().parent().html());
}

File: parse_test.py
import sys
from bs4 import BeautifulSoup

def main():
    with open('bot/woningaanbod_dump.html', 'r', encoding='utf-8') as f:
        html = f.read()
    
    soup = BeautifulSoup(html, 'html.parser')
    
    links = soup.find_all('a', href=lambda href: href and 'PublicatieId=' in href)
    for i, link in enumerate(links[:5]):
        print(f"--- Link {i} ---")
        print(link.get_text(separator=' | ', strip=True))

if __name__ == '__main__':
    main()
